presto-spi/src/main/java/io/prestosql/spi/type/Timestamps.java

```diff
@@ -43,16 +44,53 @@
 
     public static final int MILLISECONDS_PER_SECOND = 1_000;
     public static final int MICROSECONDS_PER_SECOND = 1_000_000;
+    public static final int NANOSECONDS_PER_MILLISECOND = 1_000_000;
+    public static final long NANOSECONDS_PER_DAY = 1_000_000_000L * 60 * 60 * 24;
+    public static final int PICOSECONDS_PER_NANOSECOND = 1_000;
     public static final int PICOSECONDS_PER_MICROSECOND = 1_000_000;
     public static final int PICOSECONDS_PER_MILLISECOND = 1_000_000_000;
+    public static final long PICOSECONDS_PER_SECOND = 1_000_000_000_000L;
+    public static final long PICOSECONDS_PER_MINUTE = PICOSECONDS_PER_SECOND * 60;
+    public static final long PICOSECONDS_PER_HOUR = PICOSECONDS_PER_MINUTE * 60;
+    public static final long PICOSECONDS_PER_DAY = PICOSECONDS_PER_HOUR * 24;
+    public static final long SECONDS_PER_MINUTE = 60;
+    public static final long MINUTES_PER_HOUR = 60;
 
     private Timestamps() {}
 
-    public static long round(long value, int magnitude)
+    static long round(long value, int magnitude)
     {
         return roundDiv(value, POWERS_OF_TEN[magnitude]) * POWERS_OF_TEN[magnitude];
     }
 
+    /**
+     * Rescales a value of the given precision to another precision by adding 0s or truncating.
+     */
+    static long rescale(long value, int fromPrecision, int toPrecision)
+    {
+        if (value < 0) {
+            throw new IllegalArgumentException("value must be >= 0");
+        }
+
+        if (fromPrecision <= toPrecision) {
+            value *= scaleFactor(fromPrecision, toPrecision);
+        }
+        else {
+            value /= scaleFactor(toPrecision, fromPrecision);
+        }
+
+        return value;
+    }
+
+    private static long scaleFactor(int fromPrecision, int toPrecision)
+    {
+        if (fromPrecision > toPrecision) {
+            throw new IllegalArgumentException("fromPrecision must be <= toPrecision");
```

Comment:
I think you can  skip this check. It's internal and ArrayIndexOutOfBoundsException below will be good enough

Reply:
I wanted a more user-friendly error in case there are bugs